src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -29,22 +28,32 @@ import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll, FunSuite}
 import org.scalatest.concurrent.{Eventually, PatienceConfiguration}
 import org.scalatest.time.{Minutes, Seconds, Span}
 
-import org.apache.spark.deploy.k8s.integrationtest.backend.IntegrationTestBackendFactory
-import org.apache.spark.deploy.k8s.integrationtest.backend.minikube.MinikubeTestBackend
+import org.apache.spark.deploy.k8s.integrationtest.backend.{IntegrationTestBackend, IntegrationTestBackendFactory}
 import org.apache.spark.deploy.k8s.integrationtest.config._
 
 private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll with BeforeAndAfter {
 
   import KubernetesSuite._
-  private val testBackend = IntegrationTestBackendFactory.getTestBackend()
-  private val APP_LOCATOR_LABEL = UUID.randomUUID().toString.replaceAll("-", "")
+
+  private var testBackend: IntegrationTestBackend = _
   private var sparkHomeDir: Path = _
   private var kubernetesTestComponents: KubernetesTestComponents = _
   private var sparkAppConf: SparkAppConf = _
   private var image: String = _
   private var containerLocalSparkDistroExamplesJar: String = _
+  private var appLocator: String = _
+  private var driverPodName: String = _
 
   override def beforeAll(): Unit = {
+    // The scalatest-maven-plugin gives system properties that are referenced but not set null
```

Comment:
What properties end up being "null"? Is it the ones we define in `pom.xml` but we don't give a default value?

Reply:
Yes, e.g., `spark.kubernetes.test.master`, `spark.kubernetes.test.namespace`, etc.

Reply:
We don't necessarily have to include those in the `pom.xml`.

Reply:
I tried that, but without adding them to `pom.xml`, they will have no effect.

Reply:
Strange... I'm not sure what the right approach is, given my lack of expertise in maven... so we can go with this for now.